src/term/transform/encode_pattern_matching.rs

```diff
@@ -475,7 +464,7 @@ fn get_pat_arg_count(match_path: &[Pattern]) -> (usize, usize) {
     Pattern::Ctr(_, vars) => vars.len(),
     Pattern::Num(MatchNum::Zero) => 0,
     Pattern::Num(MatchNum::Succ { .. }) => 1,
-    Pattern::Tup(..) => 2,
+    Pattern::Tup(..) => unreachable!(), // TODO: Why is this unreachable? Can this be refactored?
```

Comment:
This function is called only when adding the fields of a destructured argument as a lambda. Since we don't do this for tuples, it never gets called, but if it did, the correct value would be indeed 2.
I think you can keep it `2` but add a comment that for tuples this isn't actually called because we only destructure them at the end.